### PR TITLE
Enable zonally-periodic upsampling for non-v1 runs

### PR DESCRIPTION
**Pull request by jder**
I re-ran some comparisons in #506 and can confirm that at least at 1-degree, the bilinear upsample has an artifact at lon=0 but this one does not. Here is an example of a (doctored to increase contrast) run with the artifact in a zos snapshot:
Screenshot 2026-01-22 at 5 10 33 PM

And the next identical one with the new upsampling and no artifact:
Screenshot 2026-01-22 at 5 11 32 PM

I also replaced the upsampling with a simpler approach that does less funny stuff and is hopefully more obviously correct.

### CODE REVIEW

File: src/ocean_emulators/models/modules/blocks.py [hunk 1]
```diff
@@ -60,28 +60,20 @@ def __init__(self, upsampling: int | tuple[int, int] = 2):
...
-        # (This is just linear interpolation but pytorch requires us to use "bilinear" for
-        # 2d data.)
-        up_height = torch.nn.functional.interpolate(
-            x,
-            scale_factor=(self.scale_h, 1),
+        # Upsample with periodic padding along longitude to avoid seams and
+        # keep interpolation aligned with PyTorch's bilinear sampling grid.
+        width = x.shape[-1]
+        padded = torch.nn.functional.pad(x, (1, 1, 0, 0), mode="circular")
+        upsampled = torch.nn.functional.interpolate(
```
Comment (alxmrs) [inline src/ocean_emulators/models/modules/blocks.py:67]: While I like the striding trick on the left, I'm happy that we're delegating the sampling to this interpolate fn. It's more trustworthy for sure.

File: src/ocean_emulators/models/modules/blocks.py [hunk 2]
```diff
@@ -60,28 +60,20 @@ def __init__(self, upsampling: int | tuple[int, int] = 2):
...
-        # First upsample along latitude/height using standard "bilinear" interpolation.
-        # (This is just linear interpolation but pytorch requires us to use "bilinear" for
-        # 2d data.)
-        up_height = torch.nn.functional.interpolate(
-            x,
-            scale_factor=(self.scale_h, 1),
+        # Upsample with periodic padding along longitude to avoid seams and
+        # keep interpolation aligned with PyTorch's bilinear sampling grid.
+        width = x.shape[-1]
+        padded = torch.nn.functional.pad(x, (1, 1, 0, 0), mode="circular")
```
Comment (alxmrs) [inline src/ocean_emulators/models/modules/blocks.py:66]: A future research issue -- could be promising to look at reflective padding for the top/bottom (last two values).

**Review by alxmrs — APPROVED** [review]
LGTM! Just a few random thoughts.
